### PR TITLE
textproc/opensearch: Update to 2.11.1

### PR DESCRIPTION
**Pull request by smortex**
The basic features I use work as expected.  Some of you use more advance features, so please report failures.

### CODE REVIEW

**Comment by alonsobsd** [comment]
Hello, I was testing this version with wazuh-dashboard-plugins and it breaks completely (https://github.com/wazuh/wazuh-dashboard/issues/113). I guess it was because 2.11.0 removed angularjs. I think the best could be create a opensearch210-dashboard port for wazuh until it can works without problems with opensearch-dashboards >= 2.11.0.

**Comment by smortex** [comment]
You mean this guy was right: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/4993#issuecomment-1717986504 ? :unamused:

And people still ask [if it is acceptable to ship features in patch releases](https://github.com/opensearch-project/opensearch-build/issues/4161#issuecomment-1789764912) :roll_eyes: Good news, this is *not* what was accepted in the end.

This really gets boring…

+1 for a copy with the MAJOR + MINOR version in the port name, so `opensearch210-dashboard` or `opensearch-dashboard210` maybe more in-line with what is usually done in the ports tree.

**Comment by hackacad** [comment]
Splitting the Port at a minor version does not feel right, but theses changes break certain plugins (maybe more than just Wazuh) we should definitely go for a *210 port. It's already too late for fixing/intervening upstream...

**Comment by smortex** [comment]
I inserted a commit to copy the opensearch-dashboards port. I think it make sense to do the same with opensearch itself, right?

**Comment by hackacad** [comment]
> You mean this guy was right: [opensearch-project/OpenSearch-Dashboards#4993 (comment)](https://github.com/opensearch-project/OpenSearch-Dashboards/issues/4993#issuecomment-1717986504) ? 😒
> 
> And people still ask [if it is acceptable to ship features in patch releases](https://github.com/opensearch-project/opensearch-build/issues/4161#issuecomment-1789764912) 🙄 Good news, this is _not_ what was accepted in the end.
> 
> This really gets boring…
> 
> +1 for a copy with the MAJOR + MINOR version in the port name, so `opensearch210-dashboard` or `opensearch-dashboard210` maybe more in-line with what is usually done in the ports tree.

I see multiple new warnings after 2.10-2.11 upgrade:

```
org.opensearch.core.compress.NotXContentException: Compressor detection can only be called on some xcontent bytes or compressed xcontent bytes
        at org.opensearch.core.compress.CompressorRegistry.compressor(CompressorRegistry.java:75) ~[opensearch-core-2.11.0.jar:2.11.0]
```
        
IP Adress check has changed:

```
{"type":"log","@timestamp":"2023-11-13T15:45:32Z","tags":["error","opensearch","data"],"pid":5653,"message":"[ConnectionError]: connect ECONNREFUSED 127.0.0.1:9200"}
```

when using `server.host: 0.0.0.0` in a jail (this was just an upgrade, so behavior has definitely changed).

None if these looks like a major issue, but I'd like to do some more testing before pushing this commit.

**Comment by smortex** [comment]
> [...]
> None if these looks like a major issue, but I'd like to do some more testing before pushing this commit.

Sure! I also experienced some breakage related to the listen address on Linux deployments, but did not dig in (in my case OS listening on localhost was only listening on `::1` while OSD was using `127.0.0.1` to reach OS).

I added commits to copy the existing ports with a 210 suffix. I have not tested it since I don't use wazuh.  Please report feedback if you give it a try!

**Comment by alonsobsd** [comment]
Hello, I did some modifications to opensearch*-210 ports. Please review those changes

File: textproc/opensearch-dashboards210/files/pkg-deinstall.in [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+If %%PORTNAME%%%%PKGNAMESUFFIX%% is being deleted permanently, and you do not wish to keep any
+data that was in the cluster, then you may wish to delete the
+%%WWWDIR%% directory.  This can be done by with the command:
+
+	# rm -rf %%WWWDIR%%
+
+Please ignore this if %%PORTNAME%%%%PKGNAMESUFFIX%% is being upgraded
+EOMSG
+    ;;
+esac
```
Comment (smortex) [inline textproc/opensearch-dashboards210/files/pkg-deinstall.in]: Should we not also have this for textproc/opensearch?

File: textproc/opensearch/Makefile [hunk 2]
```diff
@@ -14,7 +14,7 @@ BUILD_DEPENDS=	jna>0:devel/jna
 RUN_DEPENDS=	bash>0:shells/bash \
 		jna>0:devel/jna
 
-CONFLICTS=	opensearch13
+CONFLICTS?=	opensearch13 opensearch210
```
Comment (smortex) [inline textproc/opensearch/Makefile:17]: :+1: Yes, I forgot to update the CONFLICTS!

File: textproc/opensearch210/Makefile [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+COMMENT=        Full-text search engine for Java
+
+CONFLICTS=      opensearch13 opensearch
+
+DISTINFO=       ${.CURDIR}/distinfo
+
+MASTERDIR=      ${.CURDIR}/../../textproc/opensearch
+
+.include "${MASTERDIR}/Makefile"
```
Comment (smortex) [inline textproc/opensearch210/Makefile:13]: Good idea I think

**Review by smortex — COMMENTED** [review]
That looks good.  I think the pkg-deinstall.in was not intended for 2.10 only?

**Edit**: I can't approve this PR because I opened it :upside_down_face:

**Comment by alonsobsd** [comment]
what about remove CHROMIUM from opensearch-dashboards OPTIONS?  Really it works from any browser based on chrome. I think the best is put a note into a pkg-message file or just remove the OPTION

File: textproc/opensearch-dashboards210/files/pkg-deinstall.in (hunk 1 above)
Comment (alonsobsd) [reply textproc/opensearch-dashboards210/files/pkg-deinstall.in]: I think this file is not being included into opensearch-dashboards port/package. I don't see any SUB_FILES entry into Makefile file

**Comment by smortex** [comment]
> what about remove CHROMIUM from opensearch-dashboards OPTIONS? Really it works from any browser based on chrome. I think the best is put a note into a pkg-message file or just remove the OPTION

The option is not enabled by default, so no objection to tweak this. I personally don't use this feature :shrug:

**Review by alonsobsd — APPROVED** [review]

**Comment by smortex** [comment]
FYI, 2.11.1 is at the corner.  I guess we can wait for it before updating?

https://github.com/opensearch-project/opensearch-build/issues/4161

**Comment by alonsobsd** [comment]
> FYI, 2.11.1 is at the corner. I guess we can wait for it before updating?
> 
> [opensearch-project/opensearch-build#4161](https://github.com/opensearch-project/opensearch-build/issues/4161)

I agree. I read it breaks logstash too if I'm not mistaken. Maybe it changes on 2.11.1

**Comment by smortex** [comment]
> I agree. I read it breaks logstash too if I'm not mistaken. Maybe it changes on 2.11.1

If I understood correctly, 2.11.1 is intended to fix the regression that break logstash when compression is used in 2.10.0.  Not a user of logstash, so I am not affected by this.  I guess a few more fixes will be part of the new release.

**Review by hackacad — APPROVED** [review]

**Comment by alonsobsd** [comment]
Hello, I  have committed opensearch* update and I have added 2.10.0 version too. I did some minor modifications to opensearch ports. Now it is more opensearch friendly ( right permissions to directories and config files) 

https://cgit.freebsd.org/ports/commit/?id=c7f140c561d4ab126286d67be6b8377a35d82f04
https://cgit.freebsd.org/ports/commit/?id=1132113d67323e1489cf89af10072cfae7dc36e5

**Pull request closed without merging**